### PR TITLE
Update info mechanism, update notification and required update dialog

### PR DESCRIPTION
**Pull request by brianpeiris**

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -520,6 +523,78 @@ module.exports = async function startServer(options) {
...
+  };
+  function getDownloadUrlForCurrentPlatform(assets) {
+    const assetPlatform = nodePlatformToAssetPlatform[process.platform];
+    return assets.find(asset => asset.name.includes(assetPlatform)).downloadUrl;
+  }
+
+  const updateInfoTimeout = 2000;
+
+  async function getLatestRelease() {
+    const token = "de8cbfb4cc0281c7b731c891df431016c29b0ace";
```
Comment (brianpeiris) [inline src/server/index.js]: Github's graphql API requires a token, so I generated a personal access token with no special scopes.

File: src/server/index.js [hunk 2]
```diff
@@ -520,6 +523,78 @@ module.exports = async function startServer(options) {
...
+    const result = await fetch("https://api.github.com/graphql", {
+      timeout: updateInfoTimeout,
+      method: "POST",
+      headers: { authorization: `bearer ${token}` },
+      body: JSON.stringify({
+        query: `
+          {
+            repository(owner: "mozillareality", name: "spoke") {
```
Comment (gfodor) [inline src/server/index.js]: do we need to include a order by here or something? I'm not that familiar with this API or graphql in general

Comment (gfodor) [reply src/server/index.js]: yeah looks like there is a way to order them, i'm guessing the default is probably OK but probably should confirm https://developer.github.com/v4/enum/releaseorderfield/

Comment (gfodor) [reply src/server/index.js]: it might be release date ascending by default, so we probably need to address this

File: src/server/index.js [hunk 3]
```diff
@@ -520,6 +523,78 @@ module.exports = async function startServer(options) {
...
+    return {
+      version: release.tag.name,
+      downloadUrl: getDownloadUrlForCurrentPlatform(release.releaseAssets.nodes)
+    };
+  }
+
+  router.get("/api/update_info", koaBody(), async ctx => {
+    try {
+      // This endpoint doesn't exist yet but we query it for future use.
+      const configEndpoint = `https://${reticulumServer}/api/v1/config`;
```
Comment (gfodor) [inline src/server/index.js]: to be a bit pedantic, this should probably be `/api/v1/config/spoke`

File: src/client/ui/dialogs/UpdateRequiredDialog.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <div className={styles.dialogContainer}>
+        <Header title="Update Required" />
+        <div className={styles.updateRequiredContainer}>
+          This version of Spoke is no longer supported.<br />
```
Comment (gfodor) [inline src/client/ui/dialogs/UpdateRequiredDialog.js]: I think these forced updates will be more about bugs/security, so we can probably drop this sentence

File: src/client/ui/menus/ToolBar.js [hunk 5]
```diff
@@ -150,9 +159,25 @@ export default class ToolBar extends Component {
...
+    const { updateAvailable, updateRequired, latestVersion, downloadUrl } = this.props.editor.updateInfo;
+    const showUpdateNotification = updateAvailable && !updateRequired && !this.state.updateNotificationDismissed;
     return (
       <div className={styles.toolbar}>
         <div className={styles.logo}>spoke</div>
+        {showUpdateNotification && (
+          <div className={styles.notification}>
+            <div className={styles.message}>
+              {`Spoke ${latestVersion} is available. `}
+              <Button className={styles.download} href={downloadUrl}>
```
Comment (gfodor) [inline src/client/ui/menus/ToolBar.js]: should we target blank this, so they don't lose work?

File: src/server/index.js (hunk 3 above)
Comment (gfodor) [reply src/server/index.js]: actually this is non-pedantic since i guess this URL actually matters :)

Comment (brianpeiris) [reply src/server/index.js]: `/api/v1/configs/spoke`?

File: src/client/ui/menus/ToolBar.js (hunk 5 above)
Comment (brianpeiris) [reply src/client/ui/menus/ToolBar.js]: Firefox just starts the download. Will check Chrome.

Comment (brianpeiris) [reply src/client/ui/menus/ToolBar.js]: Chrome does the same. The response from the URL that github provides includes a content disposition header, so browsers just download the release.

**Pull request merged**